---
title: "Files"
---

```{css echo=FALSE}
code{
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
```

## Sequence Files

Sequence files come in 2 formats, [FASTQ][fq] and [FASTA][fa]. Raw reads will always come as fastq files, sometimes in compressed file formats. Some programs are written to be able to handle compressed files, but I find it's easier to decompress the files to avoid any possible hang-ups. Compressed files typically have either a `.gz` or `.zip` file extension.

### FASTQ

The fatq format always has 4 lines per read. 

  + @Read name
  + Read sequence
  + +some line I've never seen used
  + Quality scores
  
```{r engine='bash', comment='', echo=FALSE}
head -12 misc/R1.fastq
```

### FASTA

The fasta format has two lines per read.

  + \>Read name
  + Read sequence
  
```{r engine='bash', comment='', echo=FALSE}
head -8 misc/R1.fasta
```

Though technically FASTA formay is 2 lines per sequence, you may notice that the actual sequence for the second read here is on more than 1 line. Really, programs that use FASTA files check for the '>' in the read name line to denote a new sequence.. But for best practice, use 2-line format.
  
## Mapping File

These files are used to match the barcodes of multiplexed reads to their source samples.

```{r engine='bash', comment='', echo=FALSE}
head -6 misc/mapping_file.txt
```

## Reference Database

The reference database is typically in fasta format, and is used for two purposes in this pipeline. First, it is used to compare the reads to for chimera screening. Second, it is used to classify the read clusters as belonging to particular taxa.

```{r engine='bash', comment='', echo=FALSE, eval=FALSE}
head -6 misc/db.fa
```