<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Read Processing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="misc/styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">16s Amplicon Pipeline</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="step_0.html">Files</a>
</li>
<li>
  <a href="step_1.html">Shell Environment</a>
</li>
<li>
  <a href="step_2.html">Read Processing</a>
</li>
<li>
  <a href="step_3.html">Chimera Removal</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Read Processing</h1>

</div>


<div id="splitting-the-files" class="section level3">
<h3>2.1 Splitting the Files</h3>
<p>Sequencing files are very large. Running through them repeatedly with each program can take a while. Fortunately, we can exploit their properties to enable parallel processing. For most steps in the pipeline, the reads are independent. Each read has 4 lines in a fastq file. And also, in the pair-end files, the read-pairs appear in the same order in each file. If we split the files evenly, the read pairs will always be in the complementary files. With smaller files we can use <code>parallel</code> to process them simultaneously.</p>
<pre class="bash"><code>ls $RAWDAT_DIR/*_R*.fastq* &gt; $RAWDAT_DIR/raw_seq_list.txt
mkdir -p $DIR/parallel_scripts/../split_fastqs
while read SEQS; 
    do NAME=`basename $SEQS | cut -d &quot;.&quot; -f 1`; 
    mkdir -p $DIR/split_fastqs/$NAME; 
    echo &quot;split -l 1000000 $SEQS $DIR/split_fastqs/$NAME/&quot;; 
done &lt; $RAWDAT_DIR/raw_seq_list.txt &gt; $DIR/parallel_scripts/split_fastqs.sh
cat $DIR/parallel_scripts/split_fastqs.sh | parallel -j 4
ls $DIR/split_fastqs/*/* | rev | cut -d &quot;/&quot; -f 1 | sort -u | rev &gt; $DIR/seqs_list.txt</code></pre>
<p>This will create two folders in your working directory. <code>parallel_scripts</code> is just a place to store the scripts that <code>parallel</code> will run. <code>split_fastqs</code> is where we will store our smaller fastq files (notice that it is not the same directory as our raw reads). <br> Then, the files are <code>split</code> into files of <code>1000000</code> lines, or 250,000 reads.</p>
</div>
<div id="pair-end-read-assembly" class="section level3">
<h3>2.2 Pair-End Read Assembly</h3>
<p>If you have pair-end reads, reads that were sequenced from both forward and reverse directions, then you need to combine them before processing through the rest of the pipeline. Our tool of choice for this is <a href="https://github.com/neufeld/pandaseq">pandaseq</a>, which is included in RDP-tools.</p>
<pre class="bash"><code>mkdir -p $DIR/pandaseq/assembled/../stats
while read SEQS;
    do echo &quot;$PANDASEQ -T $CORES -o $OVERLAP -e $Q -N -F -d rbkfms -l $MINLENGTH -L $MAXLENGTH -f $DIR/split_fastqs/*_R1_*/$SEQS -r $DIR/split_fastqs/*_R2_*/$SEQS 1&gt; $DIR/pandaseq/assembled/$SEQS.assembled.fastq 2&gt; $DIR/pandaseq/stats/$SEQS.assembled.stats.txt.bz2&quot;;
done &lt; $DIR/seqs_list.txt &gt; $DIR/parallel_scripts/pandaseq.sh
cat $DIR/parallel_scripts/pandaseq.sh | parallel -j 10
find $DIR/pandaseq/assembled -type f -size 0 -exec rm {} +
ls $DIR/pandaseq/assembled/* | rev | cut -d &quot;/&quot; -f 1 | rev | cut -d &quot;.&quot; -f 1 &gt; $DIR/seqs_list.txt</code></pre>
</div>
<div id="read-quality-filter" class="section level3">
<h3>2.3 Read Quality Filter</h3>
<p>As an extra measure of quality control, we can use the RDP <code>SeqFilters</code> program to check the pandaseq outputs for quality reads. This is additionally important if you did not have to assemble pair-end reads befor this.</p>
<pre class="bash"><code>mkdir -p $DIR/quality_check/seqs_$Q/../chimera_removal/../final_seqs
while read SEQS;
    do echo &quot;java -jar $RDP/RDPTools/SeqFilters.jar -Q $Q -s $DIR/pandaseq/assembled/$SEQS.assembled.fastq -o $DIR/quality_check/seqs_$Q -O $SEQS; 
python $SCRIPTS/fastq_to_fasta.py $DIR/quality_check/seqs_$Q/$SEQS/NoTag/NoTag_trimmed.fastq $DIR/quality_check/seqs_$Q/$SEQS/$SEQS.fa&quot;;
done &lt; $DIR/seqs_list.txt &gt; $DIR/parallel_scripts/qc.sh
cat $DIR/parallel_scripts/qc.sh | parallel -j 20 --delay 2
cat $DIR/quality_check/seqs_25/*/*.fa &gt; $DIR/quality_check/chimera_removal/all_combined_q25.fa</code></pre>
</div>


<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
