<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Files</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="misc/styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">16s Amplicon Pipeline</a>
</li>
<li>
  <a href="step_0.html">Files</a>
</li>
<li>
  <a href="step_1.html">Shell Environment</a>
</li>
<li>
  <a href="step_2.html">Read Processing</a>
</li>
<li>
  <a href="step_3.html">Chimera Removal</a>
</li>
<li>
  <a href="step_4.html">OTU Calling</a>
</li>
<li>
  <a href="hpcc.html">HPCC</a>
</li>
<li>
  <a href="pipeline.html">Pipeline</a>
</li>
<li>
  <a href="help.html">Help</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Files</h1>

</div>


<style type="text/css">
code{
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
</style>
<p><br></p>
<div id="sequence-files" class="section level2">
<h2>Sequence Files</h2>
<p>Sequence files come in 2 formats, [FASTQ][fq] and [FASTA][fa]. Raw reads will always come as fastq files, sometimes in compressed file formats. Some programs are written to be able to handle compressed files, but I find it’s easier to decompress the files to avoid any possible hang-ups. Compressed files typically have either a <code>.gz</code> or <code>.zip</code> file extension. <br><br></p>
<div id="fastq" class="section level3">
<h3>FASTQ</h3>
<p>The fatq format always has 4 lines per read.</p>
<ul>
<li><span class="citation">@Read</span> name</li>
<li>Read sequence</li>
<li>+some line I’ve never seen used</li>
<li>Quality scores</li>
</ul>
<pre><code>@M04771:122:000000000-B62C2:1:1101:13253:1557 1:N:0:0
TCCTTTTTTTTTTCTCTTTTTTCTTCTTTCCTTTTCTTCCCTCTTTCTTCTTCTTTTTTTTCCTTCTTTTTTTCCCTCCCTTTTCTTCCCTTTTTTTTTTCCTTTTTTCCTTTTTTTCTTTTTTTTTTTTTTTTTTTTTTTTTTTTCTTTT
+
&gt;&gt;&gt;11311&gt;@00A012BEF1AABEE22DA21DDFF2BF210B0B112DA2DF2DB111&gt;&gt;E01@@11@1@EE?011B0//011BBF12101111&lt;///&lt;/011?&lt;11&lt;0&lt;?1&lt;1&lt;-@0&lt;=000&lt;&lt;--------9@---9-9--99-///;9
@M04771:122:000000000-B62C2:1:1101:16150:1745 1:N:0:0
TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCGCGTAGGCGGTCTGGTCAGTCAGATGTGAAAGCCCCGGGCTCAACCTGGGAATTGCATTTGATACTGCCAGACTCGAGTGTGGTAGAGGGAAGTGGAATTCCGAGT
+
AAAA?1AD?AAA1FGGGGE?0BGFF00FFGAGFFGEEGGAGHFEA/AE&gt;EEEBEGE/E00BGDGG&gt;1B1BFGGFFFFECEE/&lt;ECGC1CG&lt;F00FA1?FDGBFGG&gt;FFFHHG1&lt;110?F1?.CGA&gt;1AFFFBGH?.&lt;AGBF0DDC&lt;G/.:C
@M04771:122:000000000-B62C2:1:1101:14494:1756 1:N:0:0
TACGGAGGGGGTTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCGCGTAGGCGGATTATTCAGTCAGAGGTGAAATCCCAGGGCTCAACCCTGGAACTGCCTTTGATACTGCTAGTCTTGAGTTCGAGAGAGGTGAGTGGAATTCCGAGT
+
A1AA11AD@1&gt;&gt;AAGGEGEEFFHFEC/FGFBGFFGEEGC?/GEE&gt;/&gt;EEEEGFEE/&gt;E12BF2GGD1B11?GHGB&gt;BGCC11/CCHE11G?&lt;00?011F&lt;FGGGH1??1FH11F11&gt;GHHF11GGD..0.@.?CGA0.DDEEC;C:C/.:;</code></pre>
<p><br></p>
</div>
<div id="fasta" class="section level3">
<h3>FASTA</h3>
<p>The fasta format has two lines per read.</p>
<ul>
<li>&gt;Read name</li>
<li>Read sequence</li>
</ul>
<pre><code>&gt;M04771:122:000000000-B62C2:1:1101:13253:1557 1:N:0:0
TCCTTTTTTTTTTCTCTTTTTTCTTCTTTCCTTTTCTTCCCTCTTTCTTCTTCTTTTTTTTCCTTCTTTTTTTCCCTCCCTTTTCTTCCCTTTTTTTTTTCCTTTTTTCCTTTTTTTCTTTTTTTTTTTTTTTTTTTTTTTTTTTTCTTTT
&gt;M04771:122:000000000-B62C2:1:1101:16150:1745 1:N:0:0
TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCGCGTAGGCGGTCTGG
TCAGTCAGATGTGAAAGCCCCGGGCTCAACCTGGGAATTGCATTTGATACTGCCAGACTC
GAGTGTGGTAGAGGGAAGTGGAATTCCGAGT
&gt;M04771:122:000000000-B62C2:1:1101:14494:1756 1:N:0:0
TACGGAGGGGGTTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCGCGTAGGCGGATTATTCAGTCAGAGGTGAAATCCCAGGGCTCAACCCTGGAACTGCCTTTGATACTGCTAGTCTTGAGTTCGAGAGAGGTGAGTGGAATTCCGAGT</code></pre>
<p>Though technically FASTA formay is 2 lines per sequence, you may notice that the actual sequence for the second read here is on more than 1 line. Really, programs that use FASTA files check for the ‘&gt;’ in the read name line to denote a new sequence.. But for best practice, use 2-line format. <br><br></p>
</div>
</div>
<div id="reference-database" class="section level2">
<h2>Reference Database</h2>
<p>The reference database is typically in fasta format, and is used for two purposes in this pipeline. First, it is used to compare the reads to for chimera screening. Second, it is used to classify the read clusters as belonging to particular taxa. The most commonly used databases are <a href="https://www.arb-silva.de/" target="_blank">Silva</a>, <a href="http://greengenes.secondgenome.com/" target="_blank">GreenGenes</a>, and the <a href="https://rdp.cme.msu.edu/" target="_blank">RDP</a>.</p>
<pre><code>&gt;S000494589 uncultured bacterium; YRM60L1D06060904  Lineage=Root;rootrank;Bacteria;domain;&quot;Actinobacteria&quot;;phylum;Actinobacteria;class;Acidimicrobidae;subclass;Acidimicrobiales;order;&quot;Acidimicrobineae&quot;;suborder;Acidimicrobiaceae;family;Acidimicrobium;genus
gcggcgtgctacacatgcagtcgtacgcggtggcacaccgagtggcgaacgggtgcgtaacacgtgaggaacctaccccg
aagtgggggataacaccgggaaaccggtgctaataccgcatacgctccccggaccgcatggtccagggagcaaagcctcc
gggcgcttcgggacggcctcgcggcctatcagcttgttggtggggtaacggcccaccaaggcgacgacgggtagctggtc
tgagaggacgatcagccacactgggactgagacacggcccagactcctacgggaggcagcagtggggaatattgcgcaat
gggcgaaagcctgacgcagcaacgccgcgtggaggacgaaggccttcgggttgtaaactcctttcagcagggacgaaact
gacggtacctgcagaagaagccccggctaactacgtgccagcagccgcggtaag</code></pre>
</div>
<div id="mapping-file" class="section level2">
<h2>Mapping File</h2>
<p>These files are used to match the barcodes of multiplexed reads to their source samples. Depending on the sequencing facility there may by 2 files, a mapping file and and I file. The mapping file is used to match the barcode the sample.</p>
<pre><code>#SampleID   BarcodeSequence LinkerPrimer    Description     
ATD1    TCAGCCTCAGCC    GTGTGYCAGCMGCCGCGGTAA   E_Quiroz_plate1_A1      
ATD2    TGCAGGGAACCG    GTGTGYCAGCMGCCGCGGTAA   E_Quiroz_plate1_A2      
ATD3    CAGCGCGGCTAG    GTGTGYCAGCMGCCGCGGTAA   E_Quiroz_plate1_A3      
ATD4    ATGGACCTAGCT    GTGTGYCAGCMGCCGCGGTAA   E_Quiroz_plate1_A4      
ATD5    CTAGCCCGTTCG    GTGTGYCAGCMGCCGCGGTAA   E_Quiroz_plate1_A5      </code></pre>
<p>The I file matches the read to the barcode.</p>
<pre><code>@M04771:122:000000000-B62C2:1:1101:13253:1557 1:N:0:0
TTTTTTTTTTCT
+
1111&gt;33B3B13
@M04771:122:000000000-B62C2:1:1101:16150:1745 1:N:0:0
TGGGTTAACACA
+
&gt;111&gt;&gt;11B3B1
@M04771:122:000000000-B62C2:1:1101:14494:1756 1:N:0:0
TTTCTTTTCTTT
+
111113313B33</code></pre>
<p><br></p>
</div>


<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
